### ISSUE TITLE
Placement of 'after_save' callback before 'after_*' callbacks causes false negatives and code duplication

### ISSUE THREAD

**Issue by jerridan** [rubocop/rubocop#10240]
I have a rails model with many callbacks - including:
```ruby
after_create :trigger_created_event, unless: :skip_subscriptions
after_update :trigger_updated_event, unless: :skip_subscriptions
after_destroy :trigger_deleted_event, unless: :skip_subscriptions
```

Rubocop returned no errors.

I added an `after_save` callback to the beginning of the group:
```ruby
after_save :update_status
after_create :trigger_created_event, unless: :skip_subscriptions
after_update :trigger_updated_event, unless: :skip_subscriptions
after_destroy :trigger_deleted_event, unless: :skip_subscriptions
```

I then ran `rubocop -A`, which "corrected" many errors:
`434 offenses detected, 434 offenses corrected`

Looking at the file, I saw that some of the callbacks (outside of the above list) had been moved, with rubocop reporting that they were in the wrong order (they were not).
Other callbacks were duplicated by rubocop up to ~50 times.

I undid those changes by rubocop, and changed the ordering of the `after_*` callbacks, like so:
```ruby
after_create :trigger_created_event, unless: :skip_subscriptions
after_update :trigger_updated_event, unless: :skip_subscriptions
after_destroy :trigger_deleted_event, unless: :skip_subscriptions
after_save :update_status
```

Running `rubocop -A` again returned no errors.

## Expected behavior

I would expect that rubocop either:
a) Returned no errors, regardless of the ordering of the `after_*` callbacks
b) Returned some error saying that `after_save` needs to be at the end of the list, if that is intended

## Actual behavior

Rubocop returned >400 errors, moving some callbacks and duplicating others many times.

## Steps to reproduce the problem

1. Take an existing rails model, with existing callbacks. The more the better.
2. Include a group of `after_*` callbacks, but not an `after_save`. 
3. Ensure rubocop passes.
4. Add an `after_save` callback above the other `after_*` callbacks.
5. Run `rubocop`

## RuboCop version

```
$ [bundle exec] rubocop -V
1.22.2 (using Parser 3.0.2.0, rubocop-ast 1.12.0, running on ruby 2.7.3 x86_64-darwin19)
  - rubocop-rails 2.12.4
  - rubocop-rspec 2.5.0
```
Note: Upgrading to 1.22.3 had no effect.

**Comment by jerridan** [comment]
Closing this issue as I realized afterwards this belongs in the `rubocop-rails` repo. I've re-created the issue there: https://github.com/rubocop/rubocop-rails/issues/583
